libsolidity/analysis/NameAndTypeResolver.h

```diff
@@ -87,6 +87,11 @@ class NameAndTypeResolver: private boost::noncopyable
 	/// Should only be called during the initial resolving phase.
 	std::vector<Declaration const*> nameFromCurrentScope(ASTString const& _name, bool _includeInvisibles = false) const;
 
+	/// Resolves a name in the "enclosing" scope, but also searches parent scopes.
+	/// @returns a pointer to the declaration on success or nullptr on failure.
+	/// Should only be called during the initial resolving phase.
+	Declaration const* nameFromEnclosingScope(ASTString const& _name) const;
```

Comment:
Since this method is actually returning a declaration, the name `nameFromEnclosingScope` is a bit misleading. How about `declarationFromEnclosingScope`?

Reply:
Completely agree, I was confused also. However that is how neighbor functions are called too: `nameFromCurrentScope`, `pathFromCurrentScope`. As with comments, we might want to change them synchronously.